File: pgm/ABC_Dart/ABC_390_A.dart
// import 'dart:collection';
import 'dart:io';
// import 'dart:math';

void main() {
  final List<int> A = stdin.readLineSync()!.split(' ').map(int.parse).toList();
  const String Yes = 'Yes';
  const String No = 'No';
  const int one = 1;
  const int zero = 0;
  final List<int> indice = Iterable<int>.generate(
    A.length - one,
  ).where((i) => A[i] > A[i + one]).toList();
  if (indice.length == one) {
    final int i = indice[zero];
    final List<int> list = [...A];
    final int keep = list[i];
    list[i] = list[i + one];
    list[i + one] = keep;
    if (Iterable<int>.generate(
      list.length - one,
    ).every((i) => list[i] < list[i + one])) {
      print(Yes);
    } else {
      print(No);
    }
  } else {
    print(No);
  }
  exit(0);
}

File: pgm/ABC_Kotlin/ABC_390_A.kt
// import java.time.*

// import kotlin.math.*

const val Yes: String = "Yes"
const val No: String = "No"
const val space: String = " "
const val one: Int = 1
const val zero: Int = 0

fun main() {
    val A: List<Int> = readln().split(space).map { it.toInt() }
    val indice: List<Int> = A.indices.toList().dropLast(one).filter { A[it] > A[it + one] }
    if (indice.size == one) {
        val i = indice[zero]
        val list = A.toMutableList()
        val keep = list[i]
        list[i] = list[i + one]
        list[i + one] = keep
        if (list.zipWithNext().all { it.first <= it.second }) {
            println(Yes)
        } else {
            println(No)
        }
    } else {
        println(No)
    }
    kotlin.system.exitProcess(0)
}